keeper/query_server_vault_v2.go

```diff
@@ -0,0 +1,1417 @@
+// SPDX-License-Identifier: BUSL-1.1
+//
+// Copyright (C) 2025, NASD Inc. All rights reserved.
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file of this repository and at www.mariadb.com/bsl11.
+//
+// ANY USE OF THE LICENSED WORK IN VIOLATION OF THIS LICENSE WILL AUTOMATICALLY
+// TERMINATE YOUR RIGHTS UNDER THIS LICENSE FOR THE CURRENT AND ALL OTHER
+// VERSIONS OF THE LICENSED WORK.
+//
+// THIS LICENSE DOES NOT GRANT YOU ANY RIGHT IN ANY TRADEMARK OR LOGO OF
+// LICENSOR OR ITS AFFILIATES (PROVIDED THAT YOU MAY USE A TRADEMARK OR LOGO OF
+// LICENSOR AS EXPRESSLY REQUIRED BY THIS LICENSE).
+//
+// TO THE EXTENT PERMITTED BY APPLICABLE LAW, THE LICENSED WORK IS PROVIDED ON
+// AN "AS IS" BASIS. LICENSOR HEREBY DISCLAIMS ALL WARRANTIES AND CONDITIONS,
+// EXPRESS OR IMPLIED, INCLUDING (WITHOUT LIMITATION) WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, NON-INFRINGEMENT, AND
+// TITLE.
+
+package keeper
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"cosmossdk.io/errors"
+	sdkmath "cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"dollar.noble.xyz/v3/types"
+	vaultsv2 "dollar.noble.xyz/v3/types/vaults/v2"
+)
+
+var _ vaultsv2.QueryServer = &queryServerV2{}
+
+type queryServerV2 struct {
+	vaultsv2.UnimplementedQueryServer
+	*Keeper
+}
+
+func NewQueryServerV2(keeper *Keeper) vaultsv2.QueryServer {
+	return &queryServerV2{Keeper: keeper}
+}
+
+func (q queryServerV2) VaultStats(ctx context.Context, req *vaultsv2.QueryVaultStatsRequest) (*vaultsv2.QueryVaultStatsResponse, error) {
+	if req == nil {
+		return nil, errors.Wrap(types.ErrInvalidRequest, "request cannot be nil")
+	}
+
+	stats, err := q.buildVaultStatsEntry(ctx)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Need to wrap this error I think rather than returning an internal error the the client.